app/controllers/middleware_datasource_controller.rb

```diff
@@ -0,0 +1,37 @@
+class MiddlewareDatasourceController < ApplicationController
+  include EmsCommon
+  include ContainersCommonMixin
```

Comment:
I did blindly the same as `middleware_deployment_controller.rb`

that could explain my comment in the same file:
https://github.com/ManageIQ/manageiq/pull/8257/files#diff-e1cf771ef85fc200496cc1cce21c8e4fR23
:]